BASE/src/ServerTelemetryChannel/Implementation/TelemetrySerializer.cs

```diff
@@ -27,6 +29,33 @@ public Uri EndpointAddress
         }
 
         public virtual void Serialize(ICollection<ITelemetry> items)
+        {
+            this.HandleTelemetryException(items);
+
+            var transmission = new Transmission(this.EndpointAddress, items);
+            this.Transmitter.Enqueue(transmission);
+        }
+
+        public virtual async Task<bool> Serialize(ICollection<ITelemetry> items, bool manualFlushAsync)
+        {
+            this.HandleTelemetryException(items);
+
+            var transmission = new Transmission(this.EndpointAddress, items, manualFlushAsync);
+            
+            using (var cancellationTokenSource = new CancellationTokenSource(TimeSpan.FromSeconds(100)))
+            {
+                using (cancellationTokenSource.Token.Register(() =>
+                {
+                    transmission.SetFlushTaskCompletionSource(false);
+                }))
```

Comment:
I think this can be simplified as
```suggestion
using (cancellationTokenSource.Token.Register(() =>transmission.SetFlushTaskCompletionSource(false);))
```